qiskit/extensions/unitary.py

```diff
@@ -56,11 +58,19 @@ def __init__(self, data, label=None):
             # the object to an Operator so that we can extract the underlying
             # numpy matrix from `Operator.data`.
             data = data.to_operator().data
-        # Convert to numpy array in case not already an array
-        data = numpy.array(data, dtype=complex)
-        # Check input is unitary
-        if not is_unitary_matrix(data):
-            raise ExtensionError("Input matrix is not unitary.")
+
+        # Check if there is any unbound parameter in array
+        try:
+            # Convert to numpy array in case not already an array
+            data = numpy.array(data, dtype=complex)
+
+            # Check input is unitary
+            if not is_unitary_matrix(data):
+                raise ExtensionError("Input matrix is not unitary.")
+        except ParameterTypeError:
+            print("Type of unbound Parameter cannot be determined to ensure unitarity.")
+            pass
+
```

Comment:
I'm sorry for my not understanding before, and for the mess.  I think we should revert all the changes to `UnitaryGate`; I understand the test better now, and I recognise that `UnitaryGate` isn't the problem at all, it's that the input test is rather strange.

Reply:
Reverted changes.